sdk/core/core-http/lib/serviceClient.ts

```diff
@@ -40,6 +40,12 @@ export interface ProxySettings {
   port: number;
   username?: string;
   password?: string;
+  key?: Buffer; // key for proxy server
+  ca?: Buffer[]; // ca for proxy server
+  cert?: Buffer; // cert for proxy server
+  headers?: { [key: string]: any }; // headers specific to proxy options
+  servername?: string;
+  localAddress?: string;
```

Comment:
Looks like we want `ProxySettings` to match with `tunnel.HttpsOverHttpsOptions`.
In that case, why not use the latter directly instead of `ProxySettings`?

Reply:
Because `ProxySettings` takes in few additional properties not present on the tunnel provided interface. E.g., username, password

Reply:
we use `username` and `password` to assemble `HttpsOverHttpsOptions.proxy.proxyAuth`.  Using our own types also avoid depending on @types/tunnel in our public APIs so we could move it into devDependnecies without issues.